### ISSUE TITLE
Support for lazy component imports

### ISSUE THREAD

**Issue by FluffyDiscord** [howesteve/elegua#7]
[dynamic()](https://github.com/howesteve/elegua#dynamic) is really nice to make our code cleaner.

Yet it could be even better.

Right now every component inside `dynamic` loads at the same time when user visits the page. It would be nice to have them load on-demand, with lazy imports.  The reason behind this feature request is that once your application grows and we have dozens of routes/pages, the browser then "downloads" all components at once and the initial page load can be longer, especially on mobile networks.

The lazy import should look like this (keep in mind the `svelte:component` needs to change too to [something like this](https://svelte.dev/repl/6db375be880441299f7721b54434068f?version=3.38.1)):

```svelte
<svelte:component this={dynamic($path, [
  ['/', () => import('./Home.svelte')],
  ['/about',  () => import('./About.svelte')],
  ['/blog',  () => import('./Blog.svelte')],
  ['/blog/:post',  () => import('./Post.svelte'), {id: $params['post']}],
],  () => import('./Error.svelte'))} />
```

**Comment by howesteve** [comment]
Fair enough, I'll implement this in a next version. However, probably will have to be in a separate `dynamicAsync()` function or something like this.

**Comment by FluffyDiscord** [comment]
Any updates on this?

**Comment by howesteve** [comment]
I intend to implement this in upcoming version 3.0 which will be svelte 5 compatible, as I mentioned in #8. I'll be closing this issue when I finish it.
